tck/methods/account.go

```diff
@@ -0,0 +1,234 @@
+		AddHbarTransfer(alias, hedera.NewHbar(float64(fromAliasParams.InitialBalance))).
+		Execute(a.sdkService.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	receipt, err := resp.GetReceipt(a.sdkService.Client)
+	if err != nil {
+		return nil, err
+	}
+	var accId string
+	if receipt.Status == hedera.StatusSuccess {
+		accId = receipt.AccountID.String()
+	}
+	return &response.AccountResponse{AccountId: accId, Status: receipt.Status.String()}, nil
+}
+
+// GetAccountInfo Get info for a given accountId and return a custom object containing aggregated information about the account
+func (a *AccountService) GetAccountInfo(_ context.Context, accountId string) (*response.AccountInfoResponse, error) {
+	account, err := hedera.AccountIDFromString(accountId)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := hedera.NewAccountInfoQuery().SetAccountID(account).SetGrpcDeadline(&threeSecondsDuration).Execute(a.sdkService.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	return &response.AccountInfoResponse{AccountID: resp.AccountID.String(),
+		Balance:                       resp.Balance.String(),
+		Key:                           resp.Key.String(),
+		AccountMemo:                   resp.AccountMemo,
+		MaxAutomaticTokenAssociations: resp.MaxAutomaticTokenAssociations,
+		AutoRenewPeriod:               resp.AutoRenewPeriod}, nil
+}
+
+// UpdateAccount transaction to be associated with hedera.AccountUpdateTransaction and to be enough generic, so different
+// updates can be calling this func
+func (a *AccountService) UpdateAccount(_ context.Context, params param.UpdateAccountParams) (*response.AccountResponse, error) {
+	if params.NewPrivateKey != "" { // Proceed with UpdateAccountMemo
+		return a.UpdateAccountKey(params)
+	} else if params.Memo != "" { // Proceed with UpdateAccountKey
+		return a.UpdateAccountMemo(params)
+	}
+	return nil, nil
+}
+
+// DeleteAccount deletes a provided account by signing the transaction with the key of that account
+func (a *AccountService) DeleteAccount(_ context.Context, param param.DeleteAccountParams) (*response.AccountResponse, error) {
+	accId, _ := hedera.AccountIDFromString(param.AccountId)
+	recipientId, _ := hedera.AccountIDFromString(param.RecipientId)
+	tx, err := hedera.NewAccountDeleteTransaction().SetGrpcDeadline(&threeSecondsDuration).SetAccountID(accId).SetTransferAccountID(recipientId).FreezeWith(a.sdkService.Client)
+	if err != nil {
+		return nil, err
+	}
+	signature, _ := hedera.PrivateKeyFromString(param.AccountKey)
+
+	resp, err := tx.Sign(signature).Execute(a.sdkService.Client)
+	if err != nil {
+		return nil, err
+	}
+
+	receipt, _ := resp.GetReceipt(a.sdkService.Client)
+	return &response.AccountResponse{Status: receipt.Status.String()}, nil
+}
+
+// UpdateAccountKey updates an existing acoount id keys with provided params
+func (a *AccountService) UpdateAccountKey(params param.UpdateAccountParams) (*response.AccountResponse, error) {
```

Comment:
make private

Reply:
Not sure it should be left private since the TCK has a specified test for `UpdateAccountKey`. Not fully sure tho.